jwt/src/main/scala/it/pagopa/interop/commons/jwt/package.scala

```diff
@@ -89,6 +91,33 @@ package object jwt {
   private def getInteropRoleClaimSafe(claims: JWTClaimsSet): Option[String] =
     Try(claims.getStringClaim(roleClaim)).toOption.flatMap(Option(_))
 
+  private[jwt] def getExternalId(claims: JWTClaimsSet): Option[(String, String)] = {
+    def externalIdFromOrganizationClaim(): Try[(String, String)] = for {
+      externalIdFromClaim <- Try(claims.getJSONObjectClaim(ORGANIZATION_EXTERNAL_ID_CLAIM))
+        .flatMap(nullable => Option(nullable).toTry(MissingClaim(ORGANIZATION_EXTERNAL_ID_CLAIM)))
+      origin              <- Try(externalIdFromClaim.get(ORGANIZATION_EXTERNAL_ID_ORIGIN_CLAIM))
+        .flatMap(nullable =>
+          Option(nullable)
+            .toTry(MissingClaim(s"$ORGANIZATION_EXTERNAL_ID_CLAIM.$ORGANIZATION_EXTERNAL_ID_ORIGIN_CLAIM"))
+            .map(_.toString)
+        )
+      value               <- Try(externalIdFromClaim.get(ORGANIZATION_EXTERNAL_ID_VALUE_CLAIM))
+        .flatMap(nullable =>
+          Option(nullable)
+            .toTry(MissingClaim(s"$ORGANIZATION_EXTERNAL_ID_CLAIM.$ORGANIZATION_EXTERNAL_ID_VALUE_CLAIM"))
+            .map(_.toString)
+        )
+    } yield (origin, value)
+
+    externalIdFromOrganizationClaim() match {
+      case Success((origin, value)) => Some((origin, value))
+      case Failure(e)               => {
+        logger.warn(s"Unable to extract externalId from claims: ${e.getMessage()}")
+        None
+      }
```

Comment:
The error will be logged by the calling application
```suggestion
      case Failure(e)               => None
```